### ISSUE TITLE
Loss follows Learning Rate trend when using tf.keras.optimizers.schedules.LearningRateSchedule

### ISSUE THREAD

**Issue by suraj-maniyar** [tensorflow/tensorflow#41794]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution : CentOS Linux 7.6.1810 
- TensorFlow installed from : binary (pip)
- TensorFlow version (use command below): 2.2.0-rc3
- Python version: 3.6.4
- CUDA/cuDNN version: CUDA 10.1 / cuDNN 7.6.0
- GPU model and memory: TitanX - 12GB

**Describe the current behavior**
The loss follows the pattern of learning rate when using [tf.keras.optimizers.schedules.LearningRateSchedule](https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/LearningRateSchedule). 

Examples showing the issue : 
![learning_rate_sch](https://user-images.githubusercontent.com/16310456/88607865-4f9df580-d04e-11ea-9cbf-233fa8a67d86.png)
These are few learning rate schedulers tested on mnist with different step_size and min-max range.
In all these plots, the trend in loss is similar to that of learning rate.

**Describe the expected behavior**
The loss should be independent of learning rate trend. It should show high variance at higher learning rates and should be relatively stable with low learning rate.

**Standalone code to reproduce the issue**
Find gist [here](https://colab.research.google.com/gist/suraj-maniyar/e9669de59d0694ea6407f310d9923dc7/lrschedule.ipynb)

**Comment by Saduf2019** [comment]
@suraj-maniyar 
I ran the code shared please find the [gist here](https://colab.research.google.com/gist/Saduf2019/071eee094783c664e5ece8e827c473f5/untitled297.ipynb) please let us know if it confirms your issue.

**Comment by suraj-maniyar** [comment]
@Saduf2019 
Yes. The gist you shared confirms my issue. Thanks

**Comment by sushreebarsa** [comment]
Was able to replicate the issue in TF v2.5,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/59f7c28128ae234c0740a93dd44e40f1/untitled109.ipynb)..Thanks !

**Comment by tilakrayal** [comment]
@suraj-maniyar,
When comparing the previous gist, I was able to get the different Loss on tf-nightly(2.12.0-dev20221115). Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/8584b597df8ac3b397500554875e1837/lrschedule.ipynb) and let us know the issue persists. Thank you!
